### ISSUE TITLE
Support org-roam per project setup.

### ISSUE THREAD

**Issue by exaos** [org-roam/org-roam#2468]
### Brief Abstract

I want to use multiple repositories in org-roam mode. I want some setup similar to [FOAM](https://foambubble.github.io/foam/).

### Long Description

Currently, it's cumbersome to set up multiple org-roam directories by using `.dir-locals.el` file. Moreover, this approach often doesn't work (c.f. #2419 ). I would like it if Org-roam could automatically teat a directory as a roam project when its root contains a subdirectory named `.org-roam/`, similar to how [FOAM](https://foambubble.github.io/foam/) treats directories with a `.foam/` subdirectory.

The `.org-roam/` directory could contain the `org-roam-db` file and also other settings, such as templates.

### Proposed Implementation (if any)

None.

**Comment by dmgerman** [comment]
Have you thought of using different templates to create the files so each "project" lives in a different directory? Add to the template a filetag with the project name and make it inheritable.

 This way you can use a filter function in org-roam-node-find to go to a specific project.

**Comment by exaos** [comment]
> Have you thought of using different templates to create the files so each "project" lives in a different directory? Add to the template a filetag with the project name and make it inheritable.
> 
> This way you can use a filter function in org-roam-node-find to go to a specific project.

Good idea! But this still restricts in a certain machine. If the org-roam setup is binding to a directory (a.k.a., project), It would be convenient to sync or copy just this project to another machine. Per project setup would also be more convenient for collaboration with team members.

**Comment by dmgerman** [comment]
I would not store the database in a sharable device. This way you can have different collections of documents in different computers. As long as the project lives in a subdirectory of the main roam directory, it will be tracked. org-roam will even track symlinks.

But, you need to update your db once in a while (otherwise the database will be slightly out of date). I do it daily, using the "midnight" function).

You can even have different people writing to the same org files. Each managed by their own roam.
